### PR TITLE
Support direct invocation of cargo-watch.

### LINKED ISSUE #94: Invoking cargo-watch directly should not require duplicate `watch`
Hey,

I used cargo-watch around a year or two ago and back then I could invoke it directly in the style of ./cargo/bin/cargo-watch build. I've now noticed it requires an extra `watch` to invoke --- `./cargo/bin/cargo-watch watch build`. Is that intentional or just a bug?

Git and other CLIs with external subcommands seem to work without subcommand duplication. That is, you can execute /usr/lib/git-core/git-log and get the log, not have to duplicate `log` there. I'd say cargo-watch should follow that pattern.

Thanks!

### PR DESCRIPTION
**Pull request by jmgao**
Fixes #94.

### CODE REVIEW

**Comment by passcod** [comment]
Looks good! Thanks lots. Could you add a test for the `cargo watch` case?

**Comment by jmgao** [comment]
Writing a test for the behavior I implemented seems to be hard, because rust doesn't seem to expose a way to actually modify argv[0] when starting another process. That had me wondering how cargo does it, and oops, it looks like this patch is actually broken because cargo actually just invokes it as `cargo-watch watch ...`. It was working for me locally because I had my PATH set incorrectly, so it was using the old binary I had installed previously.

I've pushed another patch that actually works.

**Comment by passcod** [comment]
Great :) Also I've added tests in `tests/exec.rs` to do the check without doing a full run, could you amend to use that instead.

**Comment by jmgao** [comment]
Sorry about the delay, I noticed some weirdness that seemed like a bug in clap, and procrastinated in investigating it (I should have just looked at the clap issues, someone already noticed it: clap-rs/clap#1390)

File: src/args.rs [hunk 1]
```diff
@@ -148,8 +151,31 @@ pub fn parse() -> ArgMatches<'static> {
                         .help("Watch specific file(s) or folder(s)"),
                 )
                 .after_help(footnote.as_str()),
-        )
-        .get_matches();
+        );
+
+    // Allow invocation of cargo-watch with both `cargo-watch watch ARGS` (as invoked by cargo) and `cargo-watch ARGS`.
+    let mut args: VecDeque<String> = env::args().collect();
```
Comment (passcod) [inline src/args.rs]: Not much point using a VecDeque instead of a Vec here now.

**Comment by passcod** [comment]
Last thing to do is to fix the module path errors for 1.26.1 (see the travis output: https://travis-ci.org/passcod/cargo-watch/jobs/493481101) and then it'll be ready to merge!

File: src/args.rs (hunk 1 above)
Comment (jmgao) [reply src/args.rs]: Fixed.

**Pull request merged**